lib/pbench/server/db/models/tracker.py

```diff
@@ -178,6 +269,11 @@ def __init__(self, **kwargs):
             state       The current state of the dataset [UPLOADING]
         """
         Dataset._render_path(kwargs)
+        if "state" in kwargs:
+            if type(kwargs.get("state")) is not States:
+                raise DatasetBadState(
+                    f"state value \"{kwargs.get('state')!r}\" is not a States enum"
```

Comment:
OK...Python's "f-strings" quoting rules are...arcane.  Nevertheless, I would recommend the following alternative:
```
                    f'state value "{kwargs.get("state")!r}" is not a States enum'
```
as, it gets rid of the backslashes but still seems to work as required.

(I like the style of using double quotes rather than single quotes, but when you need double quotes inside the string, [using single quotes is preferable to escaping the embedded double quotes](https://www.python.org/dev/peps/pep-0008/#id25).  The fact that the braces don't create a new quoting scope strikes me as odd, and it means that you have to use double quotes in the expression, but, overall that's probably a Good Thing.)